src/azure-cli/azure/cli/command_modules/identity/_client_factory.py

```diff
@@ -4,10 +4,14 @@
 # --------------------------------------------------------------------------------------------
 
 
-def _msi_client_factory(cli_ctx, **_):
+def _msi_client_factory(cli_ctx, api_version=None, **_):
     from azure.cli.core.profiles import ResourceType
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_MSI)
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_MSI, api_version=api_version)
+
+
+def _msi_list_resources_client(cli_ctx, **_):
+    return _msi_client_factory(cli_ctx, api_version='2021-09-30-preview').user_assigned_identities
```

Comment:
Please use the latest available api-version
```suggestion
    return _msi_client_factory(cli_ctx, api_version='2022-01-31-preview').user_assigned_identities
```

Reply:
Hi @isolenov,
we CLI use the api-version `2022-01-31-preview` before upgrading and it supports `listAssociatedResources` command. Could you please help to confirm which version (`2022-01-31-preview` or `2021-09-30-preview`) should be used for `listAssociatedResources` command?

Reply:
@yanzhudd @zhoxing-ms please use latest preview version `2022-01-31-preview` for `listAssociatedResources`